### PR TITLE
Fix crossentropy loss by assuming last dimension as classes

### LINKED ISSUE #24397: tf.losses.softmax_cross_entropy does not handle sequence loss when using label smoothing
When using `tf.losses.softmax_cross_entropy` with a 3D Tensor for `labels` and `logits` (which you may do if your output is a sequence), as well as the `label_smoothing` parameter, the result is incorrect. From a cursory inspection of the python source I would guess that in [line 796](https://github.com/tensorflow/tensorflow/blob/a6d8ffae097d0132989ae4688d224121ec6d8f35/tensorflow/python/ops/losses/losses_impl.py#L796), the number of classes is assumed to be `shape(onehot_labels)[1]`, rather than `shape(onehot_labels)[-1]`. The former assumes a 2D labels tensor which is neither described by the documentation nor enforced by the method.

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Mint Tumbleweed
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v1.12.0-0-ga6d8ffae09 1.12.0
- Python version: 3.5.2
- CUDA/cuDNN version: 9.0/7.1.2
- GPU model and memory: GeForce GTX 1080 Ti 11 GB

### PR DESCRIPTION
**Pull request by felix-schneider**
Fixes #24397

### CODE REVIEW

**Comment by felix-schneider** [comment]
While I would like to see this issue resolved, I'm not going through the process of signing a document for a fix that consists of a single character edit :) . I'm sure a contributor can review this and make the edit themselves.
